File: l-umi/Week_22/BOJ_30805_사전_순_최대_공통_부분_수열.py
import sys
input = sys.stdin.readline

N = int(input())
nList = list(map(int, input().split()))
M = int(input())
mList = list(map(int, input().split()))
nIndex, mIndex = 0, 0
answer = []
number=100
while number > 0 :
    nl = nList[nIndex:]
    ml = mList[mIndex:]
    if number in nl and number in ml :
        answer.append(number)
        nIndex = nl.index(number)+nIndex+1
        mIndex = ml.index(number)+mIndex+1
    else:
        number-=1

print(len(answer))
print(' '.join(map(str, answer)))

File: yunva17/Week_22/BOJ_30805_사전순최대공통부분수열/BOJ_30805_사전순최대공통부분수열.js
let input = require('fs')
  // .readFileSync('/dev/stdin')
  .readFileSync('Week_22/BOJ_30805_사전순최대공통부분수열/input.txt')
  .toString()
  .trim()
  .split('\n');

const a = Number(input[0]);
let alist = input[1].split(' ').map(Number);

const b = Number(input[2]);
let blist = input[3].split(' ').map(Number);

const searchlist = alist.slice().sort((a, b) => b - a);

let aIndex = 0,
  bIndex = 0;
const answer = [];

searchlist.forEach((v) => {
  if (alist.includes(v) && blist.includes(v)) {
    answer.push(v);
  }

  if (aIndex < a && bIndex < b && alist[aIndex] === v && blist[bIndex] === v) {
    aIndex++;
    bIndex++;
  }
});

// 결과 출력
console.log(answer.length);
console.log(answer.join(' '));
